src/DynamoCoreWpf/ViewModels/PackageManager/PackageItemRootViewModel.cs

```diff
@@ -131,8 +131,14 @@ internal void AddChildren(PackageItemRootViewModel item)
             if (this.ChildItems.Contains(item)) return;
             this.ChildItems.Add(item);
         }
-
-        internal void AddChild(PackageItemRootViewModel elem)
+        /// <summary>
+        /// The methods is used for adding a child item to all the encountered parent folders in a nested path
+        /// and make sure all the intermediate file paths are created as separte PackageItemRootViewModel.
+        /// For example if we have a path like "\dir1\dir2\dir3" and we want to add a child item to "dir1", the method will
+        /// add "dir 3" to "dir2" and then "dir2" to "dir1".
+        /// </summary>
+        /// <param name="elem">Child item to be added.</param>
+        internal void AddChildRecursively(PackageItemRootViewModel elem)
```

Comment:
hmm - I have not stared long enough, but how does this deal with subfolders with the same name.
For example:

root/extra/folder1/foo
root/extra/folder2/foo

would these both overwrite the values in a single key in the `existingSubFolders` dictionary?

Reply:
Added another test for you